lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * @fileoverview Rule to flag numbers that will lose significant figure precision at runtime
+ * @author Jacob Moore
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "disallow literal numbers that lose precision",
+            category: "Possible Errors",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-loss-of-precision"
+        },
+        schema: [],
+        messages: {
+            noLossOfPrecision: "This number will lose precision when stored as a Number type."
+        }
+    },
+
+    create(context) {
+
+        /**
+         * Returns whether the node is number literal
+         * @param {Node} node the node literal being evaluated
+         * @returns {boolean} true if the node is a number literal
+         */
+        function isNumber(node) {
+            return typeof node.value === "number";
+        }
+
+        /**
+         * Checks whether the number is not base ten
+         * @param {ASTNode} node the node being evaluated
+         * @returns {boolean} true if the node is not in base ten
+         */
+        function isNotBaseTen(node) {
```

Comment:
Could this function be changed to `isBaseTen` instead? Having the a double negative here seems like it might make the code a bit confusing to read.